api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -114,6 +121,15 @@ message RateLimitResponse {
       // Point in time at which the quota expires.
       google.protobuf.Timestamp valid_until = 2;
     }
+
+    // The unique id that is associated with each Quota either at individual descriptor level or whole descriptor set level.
+    //
+    // For the matching policy with disjunction, for example, match: "request.headers['environment'] == 'staging' || request.headers['environment'] == 'dev'"),
+    // the request_headers action produces a distinct list of descriptors for each possible value of the ‘environment’ header even though the granted quota is same.
+    // Thus, the client will use this id information (returned from RLS server) to correctly correlate the multiple descriptors/descriptor sets that have been granted with same quota (i.e., share the same quota among multiple descriptors or descriptor sets.)
+    //
+    // If id is empty, this id field will be ignored. If quota for the same id changes (e.g. due to configuration update), the old quota will be overridden by the new one. It will also be reset when quota expires after `valid_until`.
```

Comment:
"It will also be reset when quota expires after `valid_until`."

Sorry what does this part mean? I'm not sure what this is referring to.

Reply:
It refers to the this field: 
`
oneof expiration_specifier {
      // Point in time at which the quota expires.
      google.protobuf.Timestamp valid_until = 2;
    }
`
Basically, the quota object will be invalidated when it expires, so does the id. It will be reset to the new value from the RLS response that is triggered by new RLS request.

Reply:
OK then I would say: "Shared quotas referenced by ID will still adhere to expiration after `valid_until`"

Reply:
Sure. Done.